### ISSUE TITLE
Compressed reads

### ISSUE THREAD

**Issue by sjackman** [isovic/racon#54]
This command line worked with Racon 0.5.0, but fails with Racon 1.0.0.
```
❯❯❯ racon foo.fq.gz - foo.fa foo.racon.fa
[racon::createPolisher] error: file foo.fq.gz has unsupported format extension (valid extensions: .fasta, .fa, .fastq, .fq)!
```
Has support for reading compressed reads changed with Racon 1.0.0?

**Comment by rvaser** [comment]
Unfortunately yes. Piping does not work anymore as well.

**Comment by sjackman** [comment]
That's super sad for me. Any hope of bringing back those two features? They are very useful.

**Comment by rvaser** [comment]
I'll see what I can do.

**Comment by sjackman** [comment]
My use case for streaming the alignments was simply to uncompress the PAF file. i.e.
```
gunzip -c foo.paf.gz | racon foo.fq.gz - foo.fa foo.racon.fa
```
Although streaming the alignments directly into Racon was a cool feature, it's not critical to me. Reading compressed read sequences and alignments is on the other hand quite important to me.

**Comment by sjackman** [comment]
Does reading from a named pipe work? e.g.
```bash
racon <(gunzip -c foo.fq.gz) <(gunzip -c foo.paf.gz) foo.fa foo.racon.fa
```

**Comment by rvaser** [comment]
Nope, only uncompressed files are supported. I'll see about .gz files.

**Comment by sjackman** [comment]
What's the reason for pipes no longer working? Is the stream read more than once, or does it need to seek (random access)?

**Comment by rvaser** [comment]
I intended to read the files multiple times, and implemented such parsers, but can't do it for both modes in same fashion so now files are read only once. I am not sure if I can easily integrate piping nor compressed files, will have to check.

**Comment by sjackman** [comment]
Piping should just work so long as the stream is read only once, it Racon doesn't seek the stream. I haven't yet tested it. Does Racon seek the stream?

**Comment by rvaser** [comment]
You can check https://github.com/rvaser/bioparser/blob/master/include/bioparser/bioparser.hpp to see how the parsers are implemented. I am open for suggestions about piping if it is not too much troublesome. I think that compressed data won't be a problem with zlib.

**Comment by rvaser** [comment]
Version `1.1.0` now supports compressed files with gzip!

**Comment by sjackman** [comment]
I've updated the version of Racon in Brewsci/bio to 1.1.0.

**Comment by rvaser** [comment]
Great! :)

**Comment by sjackman** [comment]
Reading compressed reads and PAF works with 1.1.0! Thank you, Robert!
